### PR TITLE
MUMMNG-1318 : Create a subreddit feed example widget

### PR DESCRIPTION
**Pull request by timlevett**
As an idea for our JSON example for our seminar, we could use subreddit, so I thought I would try it.

This is what I came up with.

I started with the `List of Content` sample widget (shown here https://tools.my.wisc.edu/widget-creator/samples). Then grabbed the subreddit feed for uwmadison. Then created an expandable reddit feed.

http://goo.gl/vHvEVD

### CODE REVIEW

**Comment by vertein** [comment]
:+1: We might have to find a way around the request limit.

File: angularjs-portal-mock-portal/src/main/webapp/web/layoutDoc [hunk 1]
```diff
@@ -18,6 +18,23 @@
...
+         "title":"/r/uwmadison",
+         "description":"The UW-Madison subreddit. We do not monitor this feed.",
+         "url":"https://www.reddit.com/r/uwmadison",
+         "iconUrl":null,
+         "faIcon":"fa-reddit",
+         "fname":"uwmadison-sub-reddit",
+         "target": "_blank",
+         "widgetURL": "/web/api/uwmadisonreddit.json",
+         "widgetType":"generic",
+         "widgetTemplate":"<div style='max-height: 210px; overflow-y: scroll' ng-init='lim=5'> <ul class='widget-list' style='padding : 0'> <li ng-repeat='item in content.data.children | limitTo:lim'> <a href='{{item.data.url}}' target='_blank'> <p class='bold'>{{item.data.title}}<span class='right'>score: {{item.data.score}}</span></p> <p>{{item.data.selftext}}</p> </a> </li> <li class='center' ng-hide='content.data.children.length <= lim' ng-click='lim=lim+5'><button class='btn btn-flat center'>Load More</button></li></ul></div><a class='btn btn-default launch-app-button' href='https://www.reddit.com/r/uwmadison' target='_blank'>Launch UW subreddit</a>",
```
Comment (apetro) [inline angularjs-portal-mock-portal/src/main/webapp/web/layoutDoc]: This incantation

```
style='max-height: 210px; overflow-y: scroll'
```

looks useful and re-usable for other Widgets in this style.  Worth pulling out into some `css`?

File: angularjs-portal-mock-portal/src/main/webapp/web/layoutDoc [hunk 2]
```diff
@@ -62,7 +79,7 @@
...
...bstring(0,2)','-paid.substring(3,5)'] | limitTo:3\" class='center'><a href='/portal/p/earnings-statement/max/earning_statement.pdf.resource.uP?pP_docId={{item.docId}}' target='_blank'><i class='fa fa-bank fa-fw'></i> {{item.paid}} Statement</a></li></ul>\n           <div ng-if='isEmpty' style='padding: 10px; font-size: 14px;'><i class='fa fa-exclamation-triangle fa-3x pull-left' style='color: #b70101;'></i><span style='color: #898989;'>We had a problem finding your statements (or you don't have any).</span></div>\n           <div style='background-color: #EAEAEA; border-radius:4px;padding:10px; margin-top:10px;'>\n              <span class='bold display-block left' style='text-align: left; padding-left: 10px; font-size: 14px;'>See all payroll information for more options:</span> \n              <ul style='text-align: left;list-style-type: disc; font-size: 12px;'>\n                 <li>See all paystubs</li>\n                 <li>Tax statements</li>\n                 <li>Update direct deposit</li>\n              </ul>\n           </div>\n        </div>\n        <a class='btn btn-default launch-app-button' href='/portal/p/earnings-statement'>See all payroll information</a>\n        ",
```
Comment (apetro) [inline angularjs-portal-mock-portal/src/main/webapp/web/layoutDoc]: Looks like an unrelated change?

**Comment by apetro** [comment]
:+1: so long as that earnings statement widget template change isn't a problem.

**Comment by timlevett** [comment]
@apetro - noticed the feed was erring and it had some extra backslashes. Weird for sure, but no biggy.

File: angularjs-portal-mock-portal/src/main/webapp/web/layoutDoc (hunk 1 above)
Comment (timlevett) [inline angularjs-portal-mock-portal/src/main/webapp/web/layoutDoc]: Could. I'll check with @jhanstra about it.

Comment (jhanstra) [inline angularjs-portal-mock-portal/src/main/webapp/web/layoutDoc]: Yes, I like it :)
